src/Physics/NuElectron/EventGen/NuEKinematicsGenerator.cxx

```diff
@@ -58,6 +62,17 @@ void NuEKinematicsGenerator::ProcessEventRecord(GHepRecord * evrec) const
   const EventGeneratorI * evg = rtinfo->RunningThread();
   fXSecModel = evg->CrossSectionAlg();
 
+  //Check if kinematics is still valid
+  //CM energy may have dropped below threshold due to initial state electron velocity
+  if(!fXSecModel->ValidKinematics(evrec->Summary())) {
```

Comment:
This valid kinematics does not make sense if we don't check we are above threshold in the function.